src/main/java/seedu/placebook/model/ModelManager.java

```diff
@@ -359,6 +361,41 @@ public void updateState(String commandName) {
         this.historyStates.addNewState(stateToUpdate);
     }
 
+    private void checkValidity(Contacts contacts, Schedule schedule) {
+        ObservableList<Appointment> appointments = schedule.getSchedule();
+        ObservableList<Person> persons = contacts.getPersonList();
+        //for (Appointment invalid : findInvalidAppointments(appointments, persons)) {
+        //    logger.fine("Deleting Invalid Appointment: " + invalid.toString());
+        //    schedule.deleteAppointment(invalid);
+        //}
+
+        for (Person person : findMissingPersons(appointments, persons)) {
+            this.contacts.addPerson(person);
+        }
+    }
+
+    private UniquePersonList findMissingPersons(ObservableList<Appointment> appointments,
+                                             ObservableList<Person> persons) {
+
+        UniquePersonList missingPersons = new UniquePersonList();
+        for (Appointment appointment : appointments) {
+            ObservableList<Person> clients = appointment.getClientList();
+            addMissingPerson(missingPersons, persons, clients);
+        }
+        return missingPersons;
+    }
+
+    private void addMissingPerson(UniquePersonList missingPersons, ObservableList<Person> persons,
+                              ObservableList<Person> clients) {
+        for (Person client : clients) {
+            if (!persons.contains(client)) {
+                if (!missingPersons.contains(client)) {
+                    missingPersons.add(client);
+                }
+            }
+        }
+    }
+
```

Comment:
Really rigorous and nice implementation!
Alternatively, maybe we can consider something like:
```
UniquePersonList newPersonList = new UniquePersonList();
for (every person in this.contacts) newPersonList.add(person);

for (every person in every appointment) {
    try    newPersonList.add(person)
    catch DulplicatePersonException e    skip that person
}

this.contacts.setPersons(newPersonList);
```
I haven't tried it out but I think it also works because the uniqueness is provided by `UniquePersonList` and we don't need to think about how to check it.